catalog/controller/extension/payment/razorpay.php

```diff
@@ -124,14 +123,15 @@ public function index()
 
                 if ((isset($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === false) or
                     ((isset($this->session->data["razorpay_order_id_" . $this->session->data['order_id']]) === true) and
-                    (($this->session->data["razorpay_order_amount"] === 0) or
-                    ($this->session->data["razorpay_order_amount"] !== $order_data["amount"]))))
+                        (($this->session->data["razorpay_order_amount"] === 0) or
+                            ($this->session->data["razorpay_order_amount"] !== $order_data["amount"]))))
                 {
                     $razorpay_order = $this->api->order->create($order_data);
 
                     $this->session->data["razorpay_order_amount"] = $order_data["amount"];
                     $this->session->data["razorpay_order_id_" . $this->session->data['order_id']] = $razorpay_order['id'];
                     $data['razorpay_order_id'] = $this->session->data["razorpay_order_id_" . $this->session->data['order_id']];
+                    $this->model_extension_payment_razorpay->addOrderForWebhook($this->session->data['order_id'], $razorpay_order['id'], 0);
```

Comment:
Why are we doing this twice? Line 108 it is already added.

Reply:
@yashgit891 it was for subscription removed as we are not handling subscription events